### PR TITLE
New Market Data API upgraded to 5.2

### PR DESCRIPTION
**Pull request by jessethouin**
Updated with master, fixed bug in MarketDataWebsocketClient

### CODE REVIEW

File: src/main/java/net/jacobpeterson/alpaca/websocket/marketdata/client/MarketDataWebsocketClient.java [hunk 1]
```diff
@@ -321,20 +321,20 @@ private void submitStreamRequest() {
...
-            Set<AlpacaStreamMessageType> alpacaStreamMessageTypes = alpacaStreamListener.getStreamMessageTypes();
+        for (MarketDataStreamListener marketDataStreamListener : listeners) {
+            // Set<MarketDataStreamMessageType> marketDataStreamMessageTypes = new HashSet<>();
 
-            // if its empty, assume they want everything
-            Set<AlpacaStreamMessageType> streamMessageTypesToAdd = alpacaStreamMessageTypes == null ? new HashSet<>() :
-                    alpacaStreamMessageTypes.stream().filter(AlpacaStreamMessageType::isAPISubscribable)
-                            .collect(Collectors.toSet());
+            marketDataStreamListener.getDataStreams().forEach((s, m) -> {
+                Set<MarketDataStreamMessageType> marketDataStreamMessageTypeSet =
```
Comment (jessethouin) [inline src/main/java/net/jacobpeterson/alpaca/websocket/marketdata/client/MarketDataWebsocketClient.java]: Should have deleted this.

File: src/main/java/net/jacobpeterson/alpaca/AlpacaAPI.java [hunk 2]
```diff
@@ -83,6 +85,9 @@
     /** The alpaca web socket client. */
     private final AlpacaWebsocketClient alpacaWebSocketClient;
 
+    /** The alpaca web socket client. */
```
Comment (jessethouin) [inline src/main/java/net/jacobpeterson/alpaca/AlpacaAPI.java]: D'oh! The data websocket client.

**Comment by Petersoj** [comment]
Hey thanks! I'll review this on the weekend.

**Comment by jessethouin** [comment]
It doesn't work, and I can't figure out why. I've stepped through with the debugger, and AbstractWebsocketClientEndpoint.onMessage never fires. It connects to the ws, but I'm not sure MarketDataWebsocketClient is even sending messages, including auth. I'll keep working on it and update here. For now, don't merge.

**Comment by jessethouin** [comment]
Ok, good to go. Like the commit message says, alpaca's webservices aren't consistent in their return message types. It may be worth raising a bug to them.

edit: https://forum.alpaca.markets/t/websocket-inconsistencies/2852/2

**Pull request merged**